tools/caffe_translator/README.md

```diff
@@ -0,0 +1,78 @@
+# Caffe Translator
+Caffe Translator is a source code translator that translates Caffe code into MXNet Python code. Note that this is different from the Caffe to MXNet model converted which is available [here](https://github.com/apache/incubator-mxnet/tree/master/tools/caffe_converter).
+
+Caffe Translator takes the training/validation prototxt ([example](https://github.com/BVLC/caffe/blob/master/examples/mnist/lenet_train_test.prototxt)) and solver prototxt ([example](https://github.com/BVLC/caffe/blob/master/examples/mnist/lenet_solver.prototxt)) as input and produces MXNet Python code as output. The translated Python code uses MXNet Symbol and Module API to build the network, reads data from LMDB files, runs training and saves the trained model using the MXNet Module API.
+
+### How to use
+
+#### Get the translator:
+Download the Caffe Translator from maven [repository](https://mvnrepository.com/artifact/org.caffetranslator/caffe-translator) or [build](build_from_source.md) from source. Java Runtime Environment (JRE) is required to run the translator.
+
+#### Translate code:
+To translate `train_val.prototxt` and `solver.prototxt` to MXNet Python code, run the following command:
+```
+java -jar caffe-translator-<version>.jar --training-prototxt <train_val_prototxt_path> \
+    --solver <solver_prototxt_path> \
+    --output-file <output_file_path>
+```
+Example:
+```
+java -jar caffe-translator-0.9.1.jar --training-prototxt lenet_train_test.prototxt \
+    --solver lenet_solver.prototxt \
+    --output-file translated_code.py
+```
+
+Here is the list of command line parameters accepted by the Caffe Translator:
+- *training-prototxt*: specifies the path to the training/validation prototxt to be translated.
+- *solver-prototxt*: specifies the path to the solver prototxt to be translated.
+- *output-file*: specifies the file to write the translated output into.
+- *params-file* (optional): specifies the .caffemodel file to initialize parameters from.
+- *custom-data-layers* (optional): Specifies a comma-separated list of types of the custom data layers used in the prototxt. The translator will use [`CaffeDataIter`](https://mxnet.incubator.apache.org/how_to/caffe.html#use-io-caffedataiter) to translate these layers to MXNet.
+
+**Note:** Translated code uses [`CaffeDataIter`](https://mxnet.incubator.apache.org/how_to/caffe.html#use-io-caffedataiter) to read from LMDB files. `CaffeDataIter` requires the number of examples in LMDB file to be specified as a parameter. You can provide this information before translation using a `#caffe2mxnet` directive like shown below:
```

Comment:
Can you link to an asset that has mnist in a lmdb format for you? There's this one: http://download.caffe2.ai/databases/mnist-lmdb.zip
Or, if we don't want to link over there, why not load this file up on our own S3?

Reply:
Linked in the second paragraph.